### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by loranazarenko**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    private final UserDetailsService userDetailService;
+
+    @Bean
+    public PasswordEncoder getPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java]: Inject via method parameters. It more suitable way for @Configuration.
```suggestion
    @Bean
    public PasswordEncoder getPasswordEncoder() {
        return new BCryptPasswordEncoder();
    }

    @Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity, UserDetailsService userDetailService) throws Exception {
```

File: src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                                auth -> auth
+                                  .requestMatchers("/auth/**", "/error", "/swagger-ui/**",
+                                    "/v3/api-docs/**")
+                                  .permitAll()
+                                  .anyRequest()
+                                  .authenticated()
+                        )
+                        .httpBasic(Customizer.withDefaults())
+                        .sessionManagement(session -> session.sessionCreationPolicy(
+                                SessionCreationPolicy.STATELESS))
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java]: why do you need STATELESS session policy until you are not implemented stateless security authorisation?
```suggestion
```

File: src/main/java/mate/academy/onlinebookstore/entity/Role.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    @Enumerated(EnumType.STRING)
+    private RoleName name;
+
+    @Override
+    public String getAuthority() {
+        return "ROLE_" + name.name();
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/entity/Role.java]: make "ROLE_" constant, for performance\memory improvements 
```suggestion
        return "ROLE_" + name.name();
```

File: src/main/java/mate/academy/onlinebookstore/entity/User.java [hunk 4]
```diff
@@ -36,4 +46,49 @@ public class User {
 
     @Column(nullable = false)
     private boolean isDeleted = false;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/entity/User.java:50]: use LAZY fetch type, it has significant advantages. 
```suggestion
    @ManyToMany(fetch = FetchType.LAZY)
```

File: src/main/java/mate/academy/onlinebookstore/validation/IsbnValidator.java [hunk 5]
```diff
@@ -7,12 +7,7 @@
 public class IsbnValidator implements ConstraintValidator<Isbn, String> {
 
     private static final String PATTERN_OF_ISBN = """
-             ^(?:ISBN
-             (?:-13)?:? )?(?=[0-9]{13}$
-             |(?=(?:[0-9]+[- ]){4})
-             [- 0-9]{17}$)97[89][- ]
-             ?[0-9]{1,5}[- ]?[0-9]+[- ]
-             ?[0-9]+[- ]?[0-9]$""";
+            \\d{9}[\\d|X]""";
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/validation/IsbnValidator.java]: there is present @ISBN annotation implemented by jakarta.validation already. 

use it to reduce code amount and simplify your code to understand

File: src/main/java/mate/academy/onlinebookstore/entity/User.java (hunk 4 above)
Comment (loranazarenko) [reply src/main/java/mate/academy/onlinebookstore/entity/User.java:50]: It's wrong

File: src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+
+@RequiredArgsConstructor
+@EnableWebSecurity
+@EnableMethodSecurity
+@Configuration
+public class SecurityConfig {
+
+    private final UserDetailsService userDetailService;
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/config/SecurityConfig.java:22]: you already inject in via method param -> remove this line
```suggestion
```

**Pull request merged**